dashboard/lib/sample_data.rb

```diff
@@ -190,12 +196,58 @@ def self.create_section(options)
         # Save progress for this level if not skipping
         if best_result
           create :user_level, user: student_user, script_id: script.id,
-            level_id: script_level.level_id, attempts: 1,
+            level_id: script_level.levels.first.id, attempts: 1,
             best_result: best_result
+
+          # Create a backing channel for this level if it's a type that needs it
+          if script_level.levels.first.channel_backed?
+            ChannelToken.find_or_create_channel_token(
+              script_level.levels.first,
+              '127.0.0.1',
+              find_or_create_storage_id_for_user_id(student_user.id),
+              {
+                hidden: true
+              }
+            )
+          end
+
+          # Roll the dice to decide whether the teacher left feedback on this level
+          feedback_rand = rng.rand(100)
+          if feedback_rand < 20
+            create :teacher_feedback,
+              student: student_user,
+              teacher: options[:teacher],
+              level: script_level.levels.first,
+              comment: tiny_lipsum
+          end
         end
 
         current_level += 1
       end
     end
   end
+
+  # For the given user id, looks up the storage id, or creates a new one if the user doesn't have one.
+  def self.find_or_create_storage_id_for_user_id(user_id)
+    environment_check!
+    row = user_storage_ids_table.where(user_id: user_id)
+    return row[:id] if row
+    user_storage_ids_table.insert(user_id: user_id)
+  end
+
+  # Helper that generates a few sentences of plausible latin-esqe text, for use as obviously
+  # fake text data.
+  def self.tiny_lipsum
+    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut" \
+    " labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco" \
+    " laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in" \
+    " voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat" \
+    " non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.".
+      split(/[.,]/).
+      sample(rng.rand(3..6)).
+      map(&:strip).
+      compact.
+      map {|s| s[0].upcase + s[1, s.length].downcase}.
```

Comment:
I think can just be `.map(&:capitalize)`
```
irb(main):001:0> "lorem ipsum".capitalize
=> "Lorem ipsum"
```

Reply:
:man_facepalming: Thank you! I had a vague sense that this existed but I couldn't find it!